pkg/gui/confirmation_panel.go

```diff
@@ -13,133 +7,265 @@ import (
 	"github.com/jesseduffield/gocui"
 )
 
-func (gui *Gui) wrappedConfirmationFunction(function func(*gocui.Gui, *gocui.View) error) func(*gocui.Gui, *gocui.View) error {
+type handler func(gui *gocui.Gui, view *gocui.View) error
+
+// createConfirmationPanel creates a new confirmation panel.
+// currentView: is used to return the focus.
+// title: sets the title of the panel.
+// prompt: what to ask the user.
+// handleConfirm: what to call on confirmation.
+// handleClose: what to call on close.
+// returns an error if something goes wrong.
+func (gui *Gui) createConfirmationPanel(currentView *gocui.View, title, prompt string, handleConfirm, handleClose handler) error {
+
+	_, err := gui.g.SetViewOnBottom("commitMessage")
+	if err != nil {
+		gui.Log.Errorf("Failed to set view on bottom at createConfirmationPanel: %s\n", err)
+		return err
+	}
+
+	gui.g.Update(func(g *gocui.Gui) error {
+
+		view, _ := gui.g.View("confirmation")
+		if view != nil {
+
+			err := gui.closeConfirmationPrompt()
+			if err != nil {
+
+				errMessage := gui.Tr.TemplateLocalize(
+					"CantCloseConfirmationPrompt",
+					Teml{
+						"error": err.Error(),
+					},
+				)
+
+				gui.Log.Errorf("%s\n", errMessage)
+				return err
+			}
+		}
+
+		confirmationView, err := gui.prepareConfirmationPanel(currentView, title, prompt)
+		if err != nil {
+			gui.Log.Errorf("Failed to prepare confirmation panel at createConfirmationPanel: %s\n", err)
+			return err
+		}
+
+		confirmationView.Editable = false
+
+		err = gui.renderString(gui.g, "confirmation", prompt)
+		if err != nil {
+			gui.Log.Errorf("Failed to render string at createConfirmationPanel: %s\n", err)
+			return err
+		}
+
+		err = gui.setConfirmationHandlers(handleConfirm, handleClose)
+		if err != nil {
+			gui.Log.Errorf("Failed to set keybindings at createConfirmationPanel: %s\n", err)
+			return err
+		}
+
+		return nil
+	})
+
+	return nil
+}
+
+// createPromptPanel creates a prompt panel which is just a confirmation panel
+// without any handlers.
+// currentView: is used to return focus.
+// title: what to call the view.
+// handleConfirm: what to do on confirmation.
+// returns an error if something goes wrong.
+func (gui *Gui) createPromptPanel(currentView *gocui.View, title string, handleConfirm handler) error {
+
+	_, err := gui.g.SetViewOnBottom("commitMessage")
```

Comment:
Why aren't we calling `onNewPopupPanel()` here, as we do on the lefthand side of this diff?

Reply:
Ah i seem to have made a mistake there, adding it back now.

Reply:
Wait, i did not make a mistake, onNewPopupPanel was removed since it only was 1 line of code in the function so I just switched it to SetViewOnBottom